roachtest/tpcc: retry prometheus query during DRT

otan opened the pull request:
Previously, the prometheus endpoint may sometimes return an EOF error. This seems sporadic / random so I'm
making it retryable.

Release justification: bug fix tests only

Resolves #74684

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/96159)
<!-- Reviewable:end -->

otan committed:
roachtest/tpcc: retry prometheus query during DRT

Previously, the prometheus endpoint may sometimes return an EOF error.
This seems sporadic / random so I'm making it retryable.

Release justification: bug fix tests only

Release note: None

ajstorm reviewed (approved):
LGTM mod a couple of nits.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96159#-:-NN2LnjcExWqL8reDjFA:bt0eatj)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @herkolategan, @otan, and @smg260)

___
*[`pkg/cmd/roachtest/tests/drt.go` line 218 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96159#-NN2LSyQ7sCRd305_izN:-NN2LSyREFmV3UMOzGCu:bocyp21) ([raw file](https://github.com/cockroachdb/cockroach/blob/193d4bd357a47fbea75fb1c59228988a3461c32f/pkg/cmd/roachtest/tests/drt.go#L218)):*
> ```Go
> 			break
> 		}
> 		l.Printf("error querying prometheus, retrying: %+v", err)
> ```

Nit: Do we want a short sleep in here?
___
*[`pkg/cmd/roachtest/tests/drt.go` line 221 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96159#-NN2LdwTElhkG7uAWDXr:-NN2LdwTElhkG7uAWDXs:b561t3) ([raw file](https://github.com/cockroachdb/cockroach/blob/193d4bd357a47fbea75fb1c59228988a3461c32f/pkg/cmd/roachtest/tests/drt.go#L221)):*
> ```Go
> 	}
> 	if len(warnings) > 0 {
> 		return nil, errors.Newf("found warnings querying prometheus: %s", warnings)
> ```

Nit: Should we just log these warnings instead of returning them?  If we return them, won't the test case fail?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
bors r=ajstorm

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/96159#-:-NN3OX2I-1xhZovInTR8:baxx542)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm, @herkolategan, and @smg260)

___
*[`pkg/cmd/roachtest/tests/drt.go` line 218 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96159#-NN2LSyQ7sCRd305_izN:-NN3OSMP1gNqvJwfaPys:b-qaiku4) ([raw file](https://github.com/cockroachdb/cockroach/blob/193d4bd357a47fbea75fb1c59228988a3461c32f/pkg/cmd/roachtest/tests/drt.go#L218)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: Do we want a short sleep in here?
</blockquote></details>

this is handled by the `retry` library we used above
___
*[`pkg/cmd/roachtest/tests/drt.go` line 221 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/96159#-NN2LdwTElhkG7uAWDXr:-NN3OPQLE9gv91T-67nZ:b-r4sxkd) ([raw file](https://github.com/cockroachdb/cockroach/blob/193d4bd357a47fbea75fb1c59228988a3461c32f/pkg/cmd/roachtest/tests/drt.go#L221)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: Should we just log these warnings instead of returning them?  If we return them, won't the test case fail?
</blockquote></details>

yes, but i'd rather know they fail then silent succeed for now.


<!-- Sent from Reviewable.io -->

otan commented:
lol bors just didn't run anythings

bors r=ajstorm
